engine/src/main/java/com/datatorrent/stram/webapp/StramWebServices.java

```diff
@@ -527,6 +567,33 @@ public JSONObject getLogicalOperator(@PathParam("operatorName") String operatorN
   }
 
   @GET
+  @Path(PATH_LOGICAL_PLAN_MODULES + "/{moduleName:.+}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public JSONObject getLogicalModule(@PathParam("moduleName") String moduleName) throws Exception
+  {
+    ModuleMeta logicalModule = dagManager.getModuleMeta(moduleName);
+    if (logicalModule == null) {
+      throw new NotFoundException();
+    }
+    LogicalModuleInfo logicalModuleInfo = dagManager.getLogicalModuleInfo(logicalModule, false);
+    return new JSONObject(objectMapper.writeValueAsString(logicalModuleInfo));
+  }
+
+  @GET
+  @Path(PATH_LOGICAL_PLAN_MODULES + "/{moduleName}/flatten")
+  @Produces(MediaType.APPLICATION_JSON)
```

Comment:
looks like the flatten implementation is not done yet since it's the same as the non-flatten one.  but the /flatten here should probably not be part of the path.  I would make flatten a QueryParam.